### PR TITLE
Fix slider ticks appearing too late

### LINKED ISSUE #1846: Timings of slider components are incorrect
Compared to stable the timings of slider ticks and repeat arrows seem to be off.

Ticks appear way too late compared to stable.
qq
Blue is stable vs Purple as osu!lazer. The red arrow points to the last tick that has just fully appeared given the respective circle positions.

~~Repeat arrows can appear before the slider snaked out to them.~~
qq

### PR DESCRIPTION
**Pull request by Aergwyn**
Based on #1953 here is an equivalent to fix SliderTicks appearing way too late.

Finally closes #1846

### CODE REVIEW

**Comment by smoogipoo** [comment]
This needs a testcase.

File: osu.Game.Rulesets.Osu/Objects/SliderTick.cs [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
     {
         public int SpanIndex { get; set; }
+        public double SliderStartTime { get; set; }
+
+        protected override void ApplyDefaultsToSelf(ControlPointInfo controlPointInfo, BeatmapDifficulty difficulty)
+        {
+            base.ApplyDefaultsToSelf(controlPointInfo, difficulty);
+
+            // SliderTicks appear earlier and earlier going further into a Slider.
+            TimePreempt = StartTime - ((StartTime - SliderStartTime) / 2 + SliderStartTime - TimeFadein * 0.66f);
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Objects/SliderTick.cs]: Shouldn't this really be `SpanStartTime` or something similar? Kinda looks a bit weird when ticks are overlapping during a repeat. This seems to be what osu-stable does.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/SliderTick.cs (hunk 1 above)
Comment (Aergwyn) [reply osu.Game.Rulesets.Osu/Objects/SliderTick.cs]: This is what I got from stable code and it seems to behave identical now. I'll look into trying SpanStartTime.

**Comment by Aergwyn** [comment]
I'll just add more ticks to tests. More testcases will come with #1960 anyways.

**Comment by peppy** [comment]
Careful, this PR seems to be rolling back `osu-resources` to a previous version.

**Comment by Aergwyn** [comment]
What? How did I even do this? Will look into that in a bit.

**Review by smoogipoo — APPROVED** [review]
lgtm

**Pull request merged**